src/turbomind/layers/sampling_layers/TopPSamplingLayer.h

```diff
@@ -30,13 +31,12 @@ class TopPSamplingLayer: public BaseSamplingLayer<T> {
     void allocateBuffer(size_t batch_size, Tensor top_k, Tensor top_p) override;
     void freeBuffer() override;
 
-    uint*    runtime_top_k_buf_ = nullptr;
-    float*   runtime_top_p_buf_ = nullptr;
-    float    runtime_max_top_p_;
-    float*   initial_top_p_buf_   = nullptr;
-    float*   top_p_decay_buf_     = nullptr;
-    float*   top_p_min_buf_       = nullptr;
-    int32_t* top_p_reset_ids_buf_ = nullptr;
+    std::vector<uint>  h_runtime_top_k;
```

Comment:
nit: The naming style is better kept consistent with the existing one, and use underscores for member variables at the end.